File: python/resources/test/jython/context.py
# Helper to access the context attributes

def getContextAttribute(name, getDefaultValue):
	value = document.container.resource.context.attributes[name]
	if value == None:
		value = getDefaultValue()

		# Note: another thread might have changed our value in the meantime.
		# We'll make sure there is no duplication.

		existing = document.container.resource.context.attributes.putIfAbsent(name, value)
		if existing != None:
			value = existing

	return value

File: ruby/applications/prudence-test/libraries/jruby/context.rb

# Helper to access the context attributes

def get_context_attribute name
	value = @prudence.resource.context.attributes[name]
	if value == nil
		value = yield

		# Note: another thread might have changed our value in the meantime.
		# We'll make sure there is no duplication.

		existing = @prudence.resource.context.attributes.put_if_absent name, value
		if existing != nil
			value = existing
		end

	end

	return value
end
